asset-transfer-private-data/chaincode-go/private_asset_transfer.go

```diff
@@ -0,0 +1,497 @@
+		}
+
+	if len(assetDeleteInput.ID) == 0 {
+		return fmt.Errorf("ID field must be a non-empty string")
+	}
+
+	valAsbytes, err := ctx.GetStub().GetPrivateData(assetCollection, assetDeleteInput.ID) //get the asset from chaincode state
+		if err != nil {
+			return fmt.Errorf("failed to read asset: %s", err.Error())
+		}
+		if valAsbytes == nil {
+			return fmt.Errorf("asset private details does not exist: %s",  assetDeleteInput.ID)
+		}
+
+	var assetToDelete Asset
+	err = json.Unmarshal([]byte(valAsbytes), &assetToDelete)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal JSON: %s", err.Error())
+	}
+
+	// delete the asset from state
+	err = ctx.GetStub().DelPrivateData(assetCollection, assetDeleteInput.ID)
+	if err != nil {
+		return fmt.Errorf("Failed to delete state:" + err.Error())
+	}
+
+	// Finally, delete private details of asset
+
+	ownerCollection, err := getCollectionName(ctx) // get owners collection
+
+	err = ctx.GetStub().DelPrivateData(ownerCollection, assetDeleteInput.ID) // delete the asset
+	if err != nil {
+			return err
+	}
+
+	return nil
+
+}
+
+// DeleteProposal can be used by a user of a rejected transfer properties to
+// remove the propsal from the asset collection and the org specific collection
+func (s *SmartContract) DeleteProposal(ctx contractapi.TransactionContextInterface) error {
+
+	transMap, err := ctx.GetStub().GetTransient()
+		if err != nil {
+			return fmt.Errorf("Error getting transient: " + err.Error())
+		}
+
+	// Asset properties are private, therefore they get passed in transient field
+	transientDeleteJSON, ok := transMap["agree_delete"]
+		if !ok {
+			return fmt.Errorf("asset to delete not found in the transient map")
+		}
+
+	type assetDelete struct {
+			ID string `json:"asset_id"`
+		}
+
+	var assetDeleteInput assetDelete
+	err = json.Unmarshal(transientDeleteJSON, &assetDeleteInput)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal JSON: %s", err.Error())
+		}
+
+	if len(assetDeleteInput.ID) == 0 {
+		return fmt.Errorf("ID field must be a non-empty string")
+	}
+
+	// delete private details of agreement
+
+	orgCollection, err := getCollectionName(ctx) // get owners collection
+
+	err = ctx.GetStub().DelPrivateData(orgCollection, assetDeleteInput.ID) // delete the asset
+	if err != nil {
+			return err
+	}
+
+	// delete transfer agreement record
+
+	tranferAgreeKey, err := ctx.GetStub().CreateCompositeKey("transferAgreement",[]string{assetDeleteInput.ID}) // create composite key
+		if err != nil {
+			return fmt.Errorf("failed to create composite key: %s", err.Error())
+		}
+
+	err = ctx.GetStub().DelState(tranferAgreeKey) // remove agreement from state
+		if err != nil {
+			return err
+		}
+
+	return nil
+
+}
+
+// Return the ID of submitting client identity. This will be used to identify
+// and verify the owner on chain
+func (s *SmartContract) ReturnID(ctx contractapi.TransactionContextInterface) (string, error) {
```

Comment:
The func name is misleading. Easily confused with return the asset ID.

On second thought, I don't see  a client app needing this chaincode func, since the owner of asset is deduced in chaincode based on caller identity. So I would suggest removing this func.

Reply:
Removed